src/actions/pages.js

```diff
@@ -84,9 +84,10 @@ export function putPage(directory, filename) {
     }
     // clear errors
     dispatch({type: ActionTypes.CLEAR_ERRORS});
-    // omit raw_content, path and empty-value keys in metadata state from front_matter
+
+    // omit raw_content, path and empty 'title' value in metadata state from front_matter
     const front_matter = _.omit(metadata, (value, key, object) => {
-      return key == 'raw_content' || key == 'path' || value == '';
+      return key == 'raw_content' || key == 'path' || (key == 'title' && value == '');
```

Comment:
because declarations like `foo: false` get interpreted as a key `foo` with `no value` and hence gets stripped out from the final Front Matter